core/src/main/scala/kafka/server/ClientQuotaManager.scala

```diff
@@ -106,7 +106,8 @@ case class QuotaEntity(quotaId: QuotaId, sanitizedUser: String, clientId: String
 class ClientQuotaManager(private val config: ClientQuotaManagerConfig,
                          private val metrics: Metrics,
                          private val quotaType: QuotaType,
-                         private val time: Time) extends Logging {
+                         private val time: Time,
+                         threadNamePrefix: Option[String] = None) extends Logging {
```

Comment:
Do we need the default argument here (and the callers)? It seems like we always want to pass the value in non-test code and maybe even in test code. It could potentially just be a String and "" can be passed when we don't want a prefix.